kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/util/InitialDelayParser.java

```diff
@@ -0,0 +1,54 @@
+package org.hypertrace.core.kafkastreams.framework.util;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import java.time.Duration;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+
+public class InitialDelayParser {
+
+  private static final String INITIAL_DELAY = "initial.delay";
+  private static final Duration INITIAL_DELAY_VERSION_CHANGE = Duration.ofMinutes(6L);
+  private static final Duration INITIAL_DELAY_NO_VERSION_CHANGE = Duration.ofMillis(0L);
+  private static final String SERVICE_VERSION = "SERVICE_VERSION";
+
+  private static final InitialDelayParser instance = new InitialDelayParser();
+
+  private InitialDelayParser() {}
+
+  public static final InitialDelayParser getInstance() {
+    return instance;
+  }
+
+  public Duration getInitialDelay(Map<String, Object> streamsConfig) {
+    Optional<String> serviceVersion = Optional.ofNullable(System.getenv(SERVICE_VERSION));
+    boolean majorVersionChange = isMajorVersionChange(serviceVersion);
+    Duration defaultInitialDelay =
+        majorVersionChange ? INITIAL_DELAY_VERSION_CHANGE : INITIAL_DELAY_NO_VERSION_CHANGE;
```

Comment:
If major version changes, we may want to make the corresponding delay also configurable (instead of hardcoded 6 min).

Reply:
It is not hard-coded. We always use initial delay if configured. Please check the test case.